core/ts/__tests__/MaciState.test.ts

```diff
@@ -744,4 +720,416 @@ describe("MaciState", function () {
+
+      const ecdhKeypair = new Keypair();
+      const sharedKey = Keypair.genEcdhSharedKey(new Keypair().privKey, coordinatorKeypair.pubKey);
+
+      const message = command.encrypt(signature, sharedKey);
+      poll.publishMessage(message, ecdhKeypair.pubKey);
+
+      expect(() => {
+        poll.processMessage(message, user1Keypair.pubKey);
+      }).to.throw("failed decryption due to either wrong encryption public key or corrupted ciphertext");
+    });
+
+    it("should throw when passed a corrupted message", () => {
+      const command = new PCommand(
+        BigInt(user1StateIndex),
+        user1Keypair.pubKey,
+        BigInt(0),
+        // voice credits spent would be this value ** this value
+        BigInt(1),
+        BigInt(1),
+        BigInt(pollId),
+      );
+
+      const signature = command.sign(user1Keypair.privKey);
+
+      const ecdhKeypair = new Keypair();
+      const sharedKey = Keypair.genEcdhSharedKey(user1Keypair.privKey, coordinatorKeypair.pubKey);
+
+      const message = command.encrypt(signature, sharedKey);
+      poll.publishMessage(message, ecdhKeypair.pubKey);
+
+      message.data[0] = BigInt(0);
+
+      expect(() => {
+        poll.processMessage(message, user1Keypair.pubKey);
+      }).to.throw("failed decryption due to either wrong encryption public key or corrupted ciphertext");
+    });
+  });
+
+  describe("processMessages", () => {
+    const maciState = new MaciState(STATE_TREE_DEPTH);
+    const pollId = maciState.deployPoll(
+      BigInt(Math.floor(Date.now() / 1000) + duration),
+      maxValues,
+      treeDepths,
+      messageBatchSize,
+      coordinatorKeypair,
+    );
+
+    const poll = maciState.polls[pollId];
+
+    const user1Keypair = new Keypair();
+    // signup the user
+    const user1StateIndex = maciState.signUp(
+      user1Keypair.pubKey,
+      voiceCreditBalance,
+      BigInt(Math.floor(Date.now() / 1000)),
+    );
+
+    it("should throw if this is the first batch and currentMessageBatchIndex is not undefined", () => {
+      const command = new PCommand(
+        BigInt(user1StateIndex),
+        user1Keypair.pubKey,
+        BigInt(0),
+        BigInt(1),
+        BigInt(0),
+        BigInt(pollId),
+      );
+
+      const signature = command.sign(user1Keypair.privKey);
+
+      const ecdhKeypair = new Keypair();
+      const sharedKey = Keypair.genEcdhSharedKey(ecdhKeypair.privKey, coordinatorKeypair.pubKey);
+
+      const message = command.encrypt(signature, sharedKey);
+
+      poll.publishMessage(message, ecdhKeypair.pubKey);
+
+      // mock
+      poll.currentMessageBatchIndex = 0;
+      expect(() => poll.processMessages(pollId)).to.throw(
+        "The current message batch index should not be defined if this is the first batch",
+      );
+      poll.currentMessageBatchIndex = undefined;
+    });
+
+    it("should succeed even if send an invalid message", () => {
+      const command = new PCommand(
+        BigInt(user1StateIndex + 1),
```

Comment:
could add description here to explain why it is `BigInt(user1StateIndex + 1)`?

Reply:
good shout